core/backend/src/test/integration/BriefcaseManager.test.ts

```diff
@@ -600,67 +600,51 @@ describe("BriefcaseManager (#integration)", () => {
 
   it("should be able to show progress when downloading a briefcase (#integration)", async () => {
     const testIModelId = await HubUtility.getTestIModelId(requestContext, HubUtility.testIModelNames.stadium);
-    requestContext.enter();
 
-    let numProgressCalls: number = 0;
+    let numProgressCalls = 0;
 
     readline.clearLine(process.stdout, 0);
     readline.moveCursor(process.stdout, -20, 0);
     const downloadProgress = (loaded: number, total: number) => {
-      const message = `${HubUtility.testIModelNames.stadium} Download Progress ... ${(loaded * 100 / total).toFixed(2)}%`;
-      process.stdout.write(message);
-      readline.moveCursor(process.stdout, -1 * message.length, 0);
-      if (loaded >= total) {
-        process.stdout.write(os.EOL);
+      if (total > 0) {
+        const message = `${HubUtility.testIModelNames.stadium} Download Progress ... ${(loaded * 100 / total).toFixed(2)}%`;
+        process.stdout.write(message);
+        readline.moveCursor(process.stdout, -1 * message.length, 0);
+        if (loaded >= total)
+          process.stdout.write(os.EOL);
+        numProgressCalls++;
       }
-      numProgressCalls++;
       return 0;
     };
 
     const args = {
       contextId: testContextId,
       iModelId: testIModelId,
-      briefcaseId: 0,
+      briefcaseId: BriefcaseIdValue.Unassigned,
       onProgress: downloadProgress,
     };
     const fileName = BriefcaseManager.getFileName(args);
-    await BriefcaseManager.deleteBriefcaseFiles(fileName, requestContext);
-
+    await BriefcaseManager.deleteBriefcaseFiles(fileName);
     const props = await BriefcaseManager.downloadBriefcase(requestContext, args);
-    requestContext.enter();
-
     const iModel = await BriefcaseDb.open(requestContext, { fileName: props.fileName });
-    requestContext.enter();
-
     await IModelTestUtils.closeAndDeleteBriefcaseDb(requestContext, iModel);
-    assert.isTrue(numProgressCalls > 19000);
+    assert.isTrue(numProgressCalls > 10);

```

Comment:
using `expect(numProgressCalls).least(11)` will tell you how many calls you actually got.
Are you guaranteed to get at least one progress call no matter how fast the download is?

Reply:
made a different fix. Also added a console.log to see what the download time is

Reply:
Looks flake-proof.

Reply:
The first may be flaky but the point its failing tell me we did not propagated error that might encounter or the test do not check for error that may have caused it.

on line 363 @scsewall  you may want to check return value of following function. 
masterSeedDb.nativeDb.saveProjectGuid(projectId)

Reply:
I still think that there is a timing problem related to setting the contextId (projectId) into the briefcase and this test is exposing it. I shouldn't even need to make that call since it is on the "seed". That was an attempt to reduce "flakiness" by setting it ahead of time...